src/OpenLoco/src/World/CompanyAi/CompanyAiPathfinding.cpp

```diff
@@ -1180,6 +1193,207 @@ namespace OpenLoco::CompanyAi
+                        const auto cost = (roadExtraBaseCost * roadIdCostFactor) / 256;
+                        totalCost += cost;
+                    }
+                }
+            }
+
+            const auto& roadPieces = World::TrackData::getRoadPiece(roadId);
+            const auto& roadPieceSeq = roadPieces[index];
+            const auto roadLoc0 = pos - World::Pos3{ Math::Vector::rotate(World::Pos2{ roadPieceSeq.x, roadPieceSeq.y }, rotation), roadPieceSeq.z };
+
+            bool hasBridge = false;
+            uint8_t bridgeType = 0xFFU;
+
+            for (auto& piece : roadPieces)
+            {
+                const auto roadLoc = roadLoc0 + World::Pos3{ Math::Vector::rotate(World::Pos2{ piece.x, piece.y }, rotation), piece.z };
+
+                auto* elRoad = getRoadElement(roadLoc, rotation, roadId, piece.index, companyId);
+                if (elRoad == nullptr)
+                {
+                    continue;
+                }
+                if (elRoad->hasBridge())
+                {
+                    hasBridge = true;
+                    bridgeType = elRoad->bridge();
+                }
+
+                bool hasLevelCrossing = false;
+                // As all level crossings will be new its always going to be currentDefaultLevelCrossingType
+                const auto levelCrossingObjId = getGameState().currentDefaultLevelCrossingType;
+
+                auto clearFunc = [&totalCost, &hasLevelCrossing](World::TileElement& el) {
+                    return clearFunction(el, totalCost, hasLevelCrossing);
+                };
+                World::TileClearance::applyClearAtStandardHeight(roadLoc, elRoad->baseZ(), elRoad->clearZ(), World::QuarterTile(elRoad->occupiedQuarter(), 0), clearFunc);
+
+                if (hasLevelCrossing)
+                {
+                    auto* levelCrossingObj = ObjectManager::get<LevelCrossingObject>(levelCrossingObjId);
+                    totalCost += Economy::getInflationAdjustedCost(levelCrossingObj->costFactor, levelCrossingObj->costIndex, 10);
+                }
+            }
+
+            if (hasBridge)
+            {
+                auto* bridgeObj = ObjectManager::get<BridgeObject>(bridgeType);
+                const auto heightCost = 0 * bridgeObj->heightCostFactor; // Why 0 probably a bug
+                const auto bridgeBaseCost = Economy::getInflationAdjustedCost(bridgeObj->baseCostFactor + heightCost, bridgeObj->costIndex, 10);
+                auto cost = (bridgeBaseCost * roadIdCostFactor) / 256;
+                if (isOnWater)
+                {
+                    cost *= 2;
+                }
+                totalCost += cost;
+            }
+
+            // if (0) // Likely another bug
+            //{
+            //     const auto tunnelBaseCost = Economy::getInflationAdjustedCost(roadObj->tunnelCostFactor, 2, 8);
+            //     auto cost = (tunnelBaseCost * roadIdCostFactor) / 256;
+            //     totalCost += cost;
+            // }
```

Comment:
This just dead code, right? Do you prefer to keep it around for now, or can this be omitted?

Reply:
Yes just dead code. You can work out how it was meant to work by looking at the track equivalent